### ISSUE TITLE
Indexing on Demand

### ISSUE THREAD

**Issue by NCIEVS** [protegeproject/lucene-search-plugin#5]
Feature request:  on first use, currently the lucene plugin doesn't index until a search is attempted. Nothing happens until the indexing is complete.  This is a request for a button/function so that the user can index on demand without the need to attempt an initial search.  There should be some visual indication that indexing is ongoing (e.g. progess bar), and on completion of the indexing, the user should be notified (or some visual indication be given) that lucene searches are now enabled.

**Comment by rsgoncalves** [comment]
I split this into 2 issues: this one for on-demand indexing functionality, and another for the progress bar #7.

**Comment by johardi** [comment]
There are 2 important features regarding the indexing:
1. The indexing happens only once, i.e., when the users do their initial search. Any subsequent search or reopening the same ontology, the search engine will reuse the prebuilt index.
2. The search engine will update its index every time the users do ontology editing in Protege.

Having these features above, imo, we don't need to clutter the GUI with another button to start the indexing. The moment users start typing their first search, it signals their demand to be able to search and the engine immediately prepares the index. I agree we need some visual feedback during the indexing process (e.g., a progress bar) and Rafael is working on that right now.

If you have further concern regarding the on-demand GUI interaction, I'd be happy to revisit the issue.

**Comment by csnyulas** [comment]
Another idea that we discussed some time ago is to start the indexing in a separate thread after the loading of ontology is finished. This would of course depend on the preferences settings and whether the index file exists already or not.

**Comment by NCIEVS** [comment]
Additional items for consideration:  a) a button would minimally add to the clutter, this tab is pretty sparse as is.  b) I think more importantly its presence reminds the user that a fresh load of the vocabulary needs to be indexed before it can be searched (it's very likely that we will refresh the vocabulary periodically in order to "squash" the history - perhaps once a week if we follow our current workflow).
